### PR TITLE
test: basic flow content

### PR DESCRIPTION
**Pull request by petrovska-petro**
`forge test -vvv`

### CODE REVIEW

**Comment by gosuto-inzasheru** [comment]
getting these warnings when compiling, maybe these could be tackled once we are ready to implement our own `SmartGardenRegistry`?
```
Warning (2018): Function state mutability can be restricted to pure
  --> contracts/lib/safe-protocol/contracts/SafeProtocolRegistry.sol:95:5:
   |
95 |     function supportsInterface(bytes4 interfaceId) external view override returns (bool) {
   |     ^ (Relevant source part starts here and spans across multiple lines).

Warning (2018): Function state mutability can be restricted to pure
  --> contracts/src/modules/BaseModule.sol:98:3:
   |
98 |   function supportsInterface(
   |   ^ (Relevant source part starts here and spans across multiple lines).
```

**Comment by petrovska-petro** [comment]
> getting these warnings when compiling?

i think is simply the compiler highligthing that in that method there is not manipulation of the blockchain state, so we could feel safe labelling them as 'pure', no?

**Comment by gosuto-inzasheru** [comment]
`console.log(safe.isModuleEnabled(address(manager)));` after `safe.enableModule(address(manager));` returns false for me!

File: contracts/src/modules/DummyModule.sol [hunk 1]
```diff
@@ -24,11 +27,43 @@ contract DummyModule is BaseModule {
...
+
+  ////////////////////////////////////////////////////////////////////////////
+  // EVENTS
+  ////////////////////////////////////////////////////////////////////////////
+
+  /// @dev use for subgraph to display basic info in ui as per `safe` basis
+  event PluginTransactionExec(address safe, address gauge, uint256 timestamp);
+
+  constructor(
+    address _manager,
```
Comment (gosuto-inzasheru) [inline contracts/src/modules/DummyModule.sol:61]: wondering if it makes sense to save the manager address to `safeConfigs` too, to keep it possible to upgrade the manager for example

(longterm change, outside the scope of this pr obviously)

**Review by gosuto-inzasheru — APPROVED** [review]
test passes!!!!!

some future steps i can think of:
- move `relayer` to the manager/registry (possibly an array of relayers)
- split `executeFromPlugin` into a queue and execute function?
- rename `DummyModule` to `HarvesterPlugin`, since it now has a clear role it fulfills

File: contracts/src/modules/DummyModule.sol [hunk 2]
```diff
@@ -24,11 +27,43 @@ contract DummyModule is BaseModule {
...
+    if (msg.sender != relayer) revert UntrustedRelayer(msg.sender);
+    _;
+  }
+
+  ////////////////////////////////////////////////////////////////////////////
+  // EVENTS
+  ////////////////////////////////////////////////////////////////////////////
+
+  /// @dev use for subgraph to display basic info in ui as per `safe` basis
+  event PluginTransactionExec(address safe, address gauge, uint256 timestamp);
```
[the remark this answers is not in the document]
Comment (petrovska-petro) [reply contracts/src/modules/DummyModule.sol:58]: issue is that the [method](https://github.com/velodrome-finance/contracts/blob/main/contracts/gauges/Gauge.sol#L130) does not return the value harvested

one thing that could be done here is either read the balance aftewards of the `rewardToken` and log that in the event, but that is bit dirty i think. thinking that ultimatly, we could read this sort of [event](https://github.com/velodrome-finance/contracts/blob/main/contracts/gauges/Gauge.sol#L130)

File: contracts/src/modules/DummyModule.sol (hunk 1 above)
Comment (petrovska-petro) [reply contracts/src/modules/DummyModule.sol:61]: yes, i like the idea of having that as something that is configurable also and gives flexibility in the plugins

**Pull request merged**